airbyte/sources/base.py

```diff
@@ -329,29 +329,21 @@ def get_records(self, stream: str) -> LazyDataset:
             ) from KeyError(stream)
 
         configured_stream = configured_catalog.streams[0]
-        all_properties = set(configured_stream.stream.json_schema["properties"].keys())
+        all_properties = cast(
+            list[str], list(configured_stream.stream.json_schema["properties"].keys())
+        )
 
         def _with_logging(records: Iterable[dict[str, Any]]) -> Iterator[dict[str, Any]]:
             self._log_sync_start(cache=None)
             yield from records
             self._log_sync_success(cache=None)
 
-        def _with_missing_columns(records: Iterable[dict[str, Any]]) -> Iterator[dict[str, Any]]:
-            """Add missing columns to the record with null values."""
-            for record in records:
-                existing_properties_lower = lower_case_set(record.keys())
```

Comment:
This "missing columns" treatment is now included in the new "normalize_records" function.